doc/Overview/Mvux/WIP_Overview.md

```diff
@@ -0,0 +1,179 @@
+﻿---
+uid: Overview.Mvux.Overview
+---
+
+# MVUX Overview
+
+MVUX is a variation of the MVU design pattern that will also feel familiar to developers who have previously worked with MVVM.
+
+Considering a simple application scenario helps to understand what it does and the benefits gained from using it, so let's do that.
+
+Our application will show the current temperature based on an external source.
+
+That should seem simple enough. All it needs to do is put a number on the screen.
+Although this seems like an easy problem, as is often the case, there are more details to consider than may be immediately apparent.
+
+- What if the external data isn't immediately available when starting the app?
+- How to show that data is being initially loaded? Or updated?
+- What if no data is available?
+- What if there's an error in obtaining or processing the data?
+- How to keep the app responsive while requesting or updating the UI?
+- How to keep track of the selected item or items?
+- How to enable loading of partial data and paginate the result?
+- How to update data in the server and the UI in response to user actions?
+- Does the app need to periodically request new data or listen to the external source provide it?
+- How do we avoid threading or concurrency issues when handling new data in the background?
+- How do we make sure the code is testable?
+
+Individually, these questions and scenarios are simple to handle, but hopefully, they highlight that there is more to consider in even a very trivial application. Now imagine an application that you need to build, and with more complex data and UIs, the potential for complexity and the amount of required code can grow enormously.
+
+MVUX is a response to such situations and makes it easier to handle the above scenarios.  
+We'll look at the pattern and how to use it to make this app. 
+
+## What is MVUX?
+
+It stands for **M**odel, **V**iew, **U**pdate, e**X**tended.
+
+Looking at each individual element, it's easiest to start with the View.
+
+### View
+
+The **View** is the UI. You can write it with XAML, C#, or a combination of the two, much as you would when using another design pattern. You would use data binding to bind to the ViewModel and allow separating the View from ViewModel. MVUX does use a ViewModel, but in a different way than the MVVM pattern. For now let's refer to the ViewModel as just Model.
```

Comment:
We are going to have to decide on the terminology here. Are we using ViewModel to refer to the BPM and use Model instead of Model? Or we're going to refer to Model as just ViewModel and keep calling the BPM BPM?
I think we should just explain that Model in MVUX is in place of ViewModel in MVVM and keep calling it Model. The BRP should remain BRP, or something else NOT ViewModel.
After terminology is agreed on I'll change them throughout the document.
@mrlacey @nickrandolph @dr1rrb

Reply:
@weitzhandler what are "BPM" and "BRP"?

IMHO, we should not use the term "ViewModel" as it refers to the MVVM pattern.

Reply:
@dr1rrb Bindable Proxy Model. BRP my mistake, meant BPM.

Reply:
Great that's what I was expecting :) 

So to complete my answer, currently in the code the terms I tried to use are "entity" / "model" / "<bindable|generated> proxy" / "view", and I maintain that we should use "view model" at all as you are suggesting.

However my really concern is about the naming of the "entity" and I think that's why we are so confused: for me it's the "model" in MVVM and MVU. And what we are naming the "model" is the "View model" in the MVVM but has not real explicit equivalent in MVU ("update"?).

Maybe we would like to change that names to "model" / "<TBD - but NOT ViewModel>" / "BPM" / "view"

The part for which we should then determine a name is the place where we receive commands and we generate an updated model, even if I don't like to use a verb for that, it seems that the term used in MVU is "update".

To summurize:

| MVVM | MVU | Current | Proposal |
| --- | --- | --- | --- |
| Model | Model | Entity | Model |
| ViewModel | ... | Proxy | Proxy (a.k.a. the X) |
| ... | Update | Model / ViewModel | Update (TBD) |
| View | View | View | View

Reply:
At this stage I don't think we can use Model for the entity. As much as it might make sense from the perspective of aligning with MVU, it would require a rework of the way that MVUX works - it generates code for a class called XYZ**Model**, so we should refer to this class in the documentation as **Model**.
For the ViewModel (ie the generated class) we should use **Bindable Proxy**, and this can apply to proxies generated for both model and other entities. We shouldn't refer to Bindable Proxy Model; if we need to be specific then use Bindable Proxy for the Model.
Update can reference a method in the Model but could also be a method on an entity that returns a new instance of the entity (see the Calculator in  SimpleCalculator).
Use **Entity** to refer to data classes (ie the Model in MVVM/MVU)

Reply:
Before making any other changes to this document, I'd work on creating a glossary. Define what words like "entity" mean so that this PR isn't people sharing their opinions and it's documented for future consistency.

Reply:
We should work on an MVUX glossary page, but that's outside the scope of this PR. I will avoid using 'entity' or other ambiguous words, and if I do I'll clarify its meaning given the context.